.github/container/test-maxtext.sh

```diff
@@ -245,22 +245,58 @@ RUN_NAME="logdir" ## the RUN_NAME cannot be changed
 if [ -z "$DECODER_BLOCK" ]; then
```

Comment:
@kocchop 
instead of ```if [ -z "$DECODER_BLOCK" ]; then ...```
It could be something like:
```if [ ! -z "$DECODER_BLOCK" ]; then DECODER_ARG="decoder_block=${DECODER_BLOCK}"; fi
RUN_SETTINGS="MaxText/train.py \
        MaxText/configs/base.yml \
        run_name=${RUN_NAME} \ 
        ${DECODER_ARG} \ 
       ....
```


This remove the duplicate code and make clear this is the only different in both condition.
Right now, it isn't clear if there is other diff then this one.